Automattic-Tracks-iOS/TracksService.m

```diff
@@ -199,6 +219,31 @@ - (void)didBecomeActive:(NSNotification *)notification
 }
 
 
+- (void)reachabilityChanged:(NSNotification *)notification
+{
+    Reachability *reachability = (Reachability *)notification.object;
+    
+    // Because the containing app may already use Reachability, limit this to ours only.
+    if (reachability != self.reachability) {
+        return;
+    }
+
+    self.deviceInformation.isWiFiConnected = reachability.isReachableViaWiFi;
+    
+    if (reachability.isReachable == YES && self.isHostReachable == NO) {
```

Comment:
Please, *beware*: we had [a severe bug in Simperium](https://github.com/Simperium/simperium-ios/issues/287), which took blood to catch with the debugger, and caused lots of Out of Sync scenarios.

Reachability, when initialized to check networking against a specific domain, is not reliable (there was even a demo somewhere showing up corner cases, with Apple's own sample code).

Their recommendation (WWDC's labs) was to [initialize for Internet Connection check](https://github.com/Simperium/simperium-ios/blob/develop/Simperium/Simperium.m#L84), instead of host checks, and... just [verify the status of the currentReachability](https://github.com/Simperium/simperium-ios/blob/develop/Simperium/Simperium.m#L233) flag.